src/views/htmlcontent/src/js/components/app.component.ts

```diff
@@ -538,23 +538,36 @@ export class AppComponent implements OnInit, AfterViewChecked {
      */
     xmlLinkHandler = (cellRef: string, row: number, dataContext: JSON, colDef: any) => {
         const self = this;
-        let value = dataContext[colDef.field];
-        $(cellRef).children('.xmlLink').click(function(): void {
-            self.dataService.openLink(value, colDef.name, 'xml');
-        });
+        self.handleLink(cellRef, row, dataContext, colDef, 'xml');
     }
 
     /**
      * Add handler for clicking on json link
      */
     jsonLinkHandler = (cellRef: string, row: number, dataContext: JSON, colDef: any) => {
         const self = this;
-        let value = dataContext[colDef.field];
+        self.handleLink(cellRef, row, dataContext, colDef, 'json');
+    }
+
+    private handleLink(cellRef: string, row: number, dataContext: JSON, colDef: any, linkType: string): void {
+        const self = this;
+        let value = self.getCellValueString(dataContext, colDef);
         $(cellRef).children('.xmlLink').click(function(): void {
-            self.dataService.openLink(value, colDef.name, 'json');
+            self.dataService.openLink(value, colDef.name, linkType);
         });
     }
 
+    private getCellValueString(dataContext: JSON, colDef: any): string {
+        let returnVal = '';
+        let value = dataContext[colDef.field];
```

Comment:
This will throw if `dataContext` is undefined. This would occur if a cell was `null`. I'm not sure if it's even possible for cells to be `null` entries, however. Still might be good to check.